### ISSUE TITLE
Potential issue in chrono split in our implementation

### ISSUE THREAD

**Issue by loomlike** [recommenders-team/recommenders#146]
One of reason of using chrono split is to prevent models to learn from future data, i.e. use train samples from the events prior to the test samples'.

Our current implementation of chrono-split does groupby user (or item) and then divides chronologically for each group, which will introduce 'learn from future data' issue to some algorithms that utilize information across users (or items), like SAR item co-occurrence matrix.

E.g. let's say a user A's data is all from 2015 and B's data is all from 2018.
When we group by user and split, 2018 data can be included in training set while 2015 data can be included in test set and SAR will use those mixed information in calculating co-occurrence matrix.
(I guess for user-item affinity matrix alone, current implementation is perfectly fine).
Same thing could happen to items, e.g. some items are really popular in 2018 and some are in 2015, etc.

The simpler way (and I guess the expected behavior) of chrono split is just sort data by timestamp all together and split by ratio. After then, filter out cold users / items if needed.
This will drop larger amount of data than current implementation,
and also will drop the accuracy, especially for trend-sensitive recommendation (like movies or musics, as people may prefer newer-released items),  but those are expected behavior of this kind of problems.

Any thoughts?
